pilot/helpers/agents/TechLead.py

```diff
@@ -55,4 +49,47 @@ def create_development_plan(self):
             "development_plan": self.project.development_plan, "app_data": generate_app_data(self.project.args)
         })
 
-        return
\ No newline at end of file
+        return
+
+    def create_feature_plan(self, feature_description):
+        self.convo_feature_plan = AgentConvo(self)
+        previous_features = get_features_by_app_id(self.project.args['app_id'])
+
+        llm_response = self.convo_feature_plan.send_message('development/feature_plan.prompt',
+            {
+                "name": self.project.args['name'],
+                "app_type": self.project.args['app_type'],
+                "app_summary": self.project.project_description,
+                "clarifications": self.project.clarifications,
+                "user_stories": self.project.user_stories,
+                "user_tasks": self.project.user_tasks,
+                "technologies": self.project.architecture,
+                "directory_tree": self.project.get_directory_tree(True),
+                "development_tasks": self.project.development_plan,
+                "files": self.project.get_all_coded_files(),
+                "previous_features": previous_features,
+                "feature_description": feature_description,
```

Comment:
Is this the same data that we use to create the project plan? Can we add something like `project.get_context(stage)` as I suggested in #147?

Reply:
There are minor differences but we should do that in future